src/qt/optionsmodel.cpp

```diff
@@ -151,10 +151,31 @@ void OptionsModel::Init(bool resetSettings)
     language = settings.value("language").toString();
 }
 
+/** Helper function to copy contents from one QSettings to another.
+ * By using allKeys this also covers nested settings in a hierarchy.
+ */
+static void copySettings(QSettings &dst, const QSettings &src)
+{
+    for (const QString &key: src.allKeys()) {
+        dst.setValue(key, src.value(key));
+    }
+}
+
+/** Back up a QSettings to an ini-formatted file. */
+static void backupSettings(const fs::path &filename, const QSettings &src)
+{
+    QSettings dst(GUIUtil::boostPathToQString(filename), QSettings::IniFormat);
+    qInfo() << "Backing up GUI settings to" << GUIUtil::boostPathToQString(filename);
+    copySettings(dst, src);
+}
+
 void OptionsModel::Reset()
 {
     QSettings settings;
 
+    // Backup old settings for troubleshooting
+    backupSettings(GetDataDir(true) / "guisettings.bak", settings);
```

Comment:
Tiny nit, `GetDataDir()` argument defaults to `true` so it'd be clearer to leave it out imo